nilearn/plotting/surf_plotting.py

```diff
@@ -29,10 +33,416 @@
+                      threshold=None, output_file=None):
+    """Helper function for plot_surf.
+
+    .. versionadded:: 0.8.1
+
+    This function handles surface plotting when the selected
+    engine is plotly.
+    """
+    coords, faces = load_surf_mesh(surf_mesh)
+    x, y, z = coords.T
+    i, j, k = faces.T
+    if cmap is None:
+        cmap = cold_hot
+    if surf_map is not None:
+        surf_map_faces = _compute_surf_map_faces(
+            surf_map, faces, avg_method,
+            coords.shape[0], faces.shape[0]
+        )
+        colors = colorscale(cmap, surf_map_faces, threshold)
+        vertexcolor = _get_vertexcolor(
+            surf_map_faces,
+            colors["cmap"],
+            colors["norm"],
+            colors["abs_threshold"],
+            bg_map=bg_map,
+        )
+    else:
+        vertexcolor = None
+    if colorbar:
+        mesh_3d = go.Mesh3d(x=x, y=y, z=z, i=i, j=j, k=k,
+                            intensity=surf_map_faces,
+                            colorscale=colors['colors'])
+    else:
+        mesh_3d = go.Mesh3d(x=x, y=y, z=z, i=i, j=j, k=k,
+                            vertexcolor=vertexcolor)
+    cameras_view = _set_view_plot_surf_plotly(hemi, view)
+    fig = go.Figure(data=[mesh_3d])
+    fig.update_layout(scene_camera=CAMERAS[cameras_view], **LAYOUT)
+    if output_file is not None:
+        fig.write_image(output_file)
+    return fig
+
+
+def _set_view_plot_surf_matplotlib(hemi, view):
+    """Helper function for plot_surf with matplotlib engine.
+
+    This function checks the selected hemisphere and view, and
+    returns elev and azim.
+    """
+    if hemi == 'right':
+        if view == 'lateral':
+            elev, azim = 0, 0
+        elif view == 'medial':
+            elev, azim = 0, 180
+        elif view == 'dorsal':
+            elev, azim = 90, 0
+        elif view == 'ventral':
+            elev, azim = 270, 0
+        elif view == 'anterior':
+            elev, azim = 0, 90
+        elif view == 'posterior':
+            elev, azim = 0, 270
+        else:
+            raise ValueError(f"view must be one of {VALID_VIEWS}")
+    elif hemi == 'left':
+        if view == 'medial':
+            elev, azim = 0, 0
+        elif view == 'lateral':
+            elev, azim = 0, 180
+        elif view == 'dorsal':
+            elev, azim = 90, 0
+        elif view == 'ventral':
+            elev, azim = 270, 0
+        elif view == 'anterior':
+            elev, azim = 0, 90
+        elif view == 'posterior':
+            elev, azim = 0, 270
+        else:
+            raise ValueError(f"view must be one of {VALID_VIEWS}")
+    else:
+        raise ValueError(f"hemi must be one of {VALID_HEMISPHERES}")
+    return elev, azim
+
+
+def _compute_surf_map_faces(surf_map, faces, avg_method, n_vertices,
+                            face_colors_size):
+    """Helper function for plot_surf.
+
+    This function computes the surf map faces using the
+    provided averaging method.
```

Comment:
```suggestion
    provided averaging method.
    This method is called exclusively when using matplotlib,
    since it only supports plotting face-colour maps
    and not vertex-colour maps.
```